components/ws-manager-api/go/crd/v1/workspace_types.go

```diff
@@ -0,0 +1,226 @@
+	WorkspacePhaseStopped      WorkspacePhase = "stopped"
+)
+
+type WorkspaceConditions struct {
+	// Deployed indicates if a workspace pod is currently deployed.
+	// If this condition is false, there is no means for the user to alter the workspace content.
+	Deployed bool `json:"deployed,omitempty"`
+
+	// Failed contains the reason the workspace failed to operate. If this field is empty, the workspace has not failed.
+	Failed string `json:"failed,omitempty"`
+
+	// Timeout contains the reason the workspace has timed out. If this field is empty, the workspace has not timed out.
+	Timeout string `json:"timeout,omitempty"`
+
+	// FirstUserActivity is the time when MarkActive was first called on the workspace
+	FirstUserActivity *metav1.Time `json:"firstUserActivity,omitempty"`
+
+	// HeadlessTaskFailed indicates that a headless workspace task failed
+	HeadlessTaskFailed string `json:"headlessTaskFailed,omitempty"`
+
+	// StoppedByRequest is true if the workspace was stopped using a StopWorkspace call
+	StoppedByRequest *bool `json:"stoppedByRequest,omitempty"`
+
+	// EverReady becomes true if the workspace was ever ready to be used
+	EverReady bool `json:"everReady,omitempty"`
+}
+
+type WorkspaceResults struct {
+	// Snapshot contains a snapshot URL if a snapshot was produced prior to shutting the workspace down. This condition is only used for headless workspaces.
+	Snapshot string `json:"snapshot,omitempty"`
+
+	// HeadlessTaskFailed indicates that a headless workspace task failed
+	HeadlessTaskFailed string `json:"headlessTaskFailed,omitempty"`
+}
+
+// workspaceDisposalStatus indicates the status of the workspace diposal
+type WorkspaceDisposalStatus struct {
+	BackupComplete bool       `json:"backupComplete,omitempty"`
+	BackupFailure  string     `json:"backupFailure,omitempty"`
+	GitStatus      *GitStatus `json:"git,omitempty"`
+}
+
+type GitStatus struct {
+	// branch is branch we're currently on
+	Branch string `json:"branch,omitempty"`
+	// latest_commit is the most recent commit on the current branch
+	LatestCommit string `json:"latest_commit,omitempty"`
+	// uncommited_files is the number of uncommitted files, possibly truncated
+	UncommitedFiles []string `json:"uncommited_files,omitempty"`
+	// the total number of uncommited files
+	TotalUncommitedFiles int64 `json:"total_uncommited_files,omitempty"`
+	// untracked_files is the number of untracked files in the workspace, possibly truncated
+	UntrackedFiles []string `json:"untracked_files,omitempty"`
+	// the total number of untracked files
+	TotalUntrackedFiles int64 `json:"total_untracked_files,omitempty"`
+	// unpushed_commits is the number of unpushed changes in the workspace, possibly truncated
+	UnpushedCommits []string `json:"unpushed_commits,omitempty"`
+	// the total number of unpushed changes
+	TotalUnpushedCommits int64 `json:"total_unpushed_commits,omitempty"`
+}
+
+type WorkspaceRuntimeStatus struct {
+	NodeName string `json:"nodeName,omitempty"`
```

Comment:
Here we should use references to the Pod object, like using the [same approach as k8s Gateway](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/object_reference_types.go#L94-L133)

Reply:
This version of the rewrite still attempts to integrate the "legacy content init" using ws-daemon. We could simplify the thing if we just went for PVC straight away.

Reply:
>  if we just went for PVC straight away.

I think this is not optional ;)